### PR TITLE
fix: `function_to_constant` `get_class()` replacement

### PR DESCRIPTION
**Pull request by mvorisek**
fix #2944

repro: https://3v4l.org/j0daX#v8.2.0

also modernize `__CLASS__` usages to `self::class` as done by Rector

### CODE REVIEW

File: src/Fixer/LanguageConstruct/FunctionToConstantFixer.php [hunk 1]
```diff
@@ -167,7 +171,9 @@ private function fixFunctionCallToConstant(Tokens $tokens, int $index, int $brac
             $tokens->clearTokenAndMergeSurroundingWhitespace($i);
         }
 
-        if ($replacements[0]->isGivenKind([T_CLASS_C, T_STATIC])) {
+        if ($replacements[0]->isGivenKind([T_CLASS_C, T_STATIC])
+            || ($replacements[0]->isGivenKind(T_STRING) && 'self' === $replacements[0]->getContent())
+        ) {
```
Comment (julienfalque) [inline src/Fixer/LanguageConstruct/FunctionToConstantFixer.php:176]: According to [PER Coding Style](https://www.php-fig.org/per/coding-style/#51-if-elseif-else), the conditions in the `if` must all be on their own lines when multiline (similar to e.g. function calls). I thought we had a rule for that.

**Review by julienfalque — APPROVED** [review]
Just a small coding style issue and ready to merge IMO

File: src/Fixer/LanguageConstruct/FunctionToConstantFixer.php (hunk 1 above)
Comment (mvorisek) [reply src/Fixer/LanguageConstruct/FunctionToConstantFixer.php]: Done and fixer will be really welcomed as the "same startling line style" is quite common not only in general, but also in this repo.

**Review by Wirone — APPROVED** [review]

**Comment by Wirone** [comment]
Thank you @mvorisek 🍻

**Pull request merged**